Sources/SwiftGRPCNIO/GRPCStatus.swift

```diff
@@ -27,4 +27,19 @@ public struct GRPCStatus: Error {
   public static func unimplemented(method: String) -> GRPCStatus {
     return GRPCStatus(code: .unimplemented, message: "unknown method " + method)
   }
+
+  // These status codes are informed by: https://github.com/grpc/grpc/blob/master/doc/statuscodes.md
+  static internal let requestProtoParseError = GRPCStatus(code: .internalError, message: "could not parse request proto")
```

Comment:
Why `internal`? In general, I prefer to have more open visibility. That could help low-level consumers of this library looking to re-implement/modify parts of it without re-inventing the wheel. Let me know if you think differently.

(If we decide to keep this internal, I'd still suggest dropping the `internal` keyword.)

Reply:
> 1. Should we have a `CollectingServerErrorDelegate` (similar to `CollectingChannelHandler`) in our tests and assert that the error delegate is sent the expected of errors?

This should be pretty straightforward to do. I'll include this in the next revision.

> 2. It might be nice to provide some more error context to `ErrorDelegate`, e.g. which `Server/CallHandler/etc.` (and the corresponding method path) has caught the current error, so we know more precisely where the error occurred.

This is _potentially_ useful for us. However, with the new `enum` of error cases, each one is only thrown in one or two places so I think we can leave this for another day.

Reply:
I'm fine with keeping the `internal`, will try not to mention this in the future. I understand the value of not exposing these methods to avoid confusing users; let's keep this stuff internal until users ask to open this up, I guess.

In general, my reasoning behind making things public/open is to let users drop in at a lower level while still re-using as much of our work as possible. `internal/private` makes this much harder, but of course makes the exposed interface "cleaner". Tough trade-off. What's your stance on this? (I'm unresolving comment this to keep the discussion flowing, but no need for you to make further changes in the code related to this.)